lib/ecto/changeset/relation.ex

```diff
@@ -252,6 +251,27 @@ defmodule Ecto.Changeset.Relation do
     Changeset.change(current, changes) |> put_new_action(:update)
   end
 
+  @doc """
+  TODO
+  """
+  def on_replace(%{on_replace: :mark_as_invalid}, changeset_or_model) do
+    %{Changeset.change(changeset_or_model) | valid?: false}
```

Comment:
Only with those. We typically assume the other ones are valid (they will fail on dump). Hrm... now I am not so sure.